src/dtx/dtx_rpc.c

```diff
@@ -301,15 +301,18 @@ dtx_req_list_cb(void **args)
 					drr->drr_rank, drr->drr_tag);
 				return;
 			case -DER_EXCLUDED:
-				/* If non-leader is excluded, handle it
-				 * as 'prepared'. If other non-leaders
-				 * also 'prepared' then related DTX is
-				 * committable. Fall through.
+				/*
+				 * If non-leader is excluded, handle it as 'prepared'. If other
+				 * non-leaders are also 'prepared' then related DTX maybe still
+				 * committable or 'corrupted'. The subsequent DTX resync logic
+				 * will handle related things, see dtx_verify_groups().
+				 *
+				 * Fall through.
 				 */
 			case DTX_ST_PREPARED:
 				if (dra->dra_result == 0 ||
 				    dra->dra_result == DTX_ST_CORRUPTED)
```

Comment:
DTX_ST_CORRUPTED will be turned into DTX_ST_PREPARED as well?

Reply:
Yes, generally, if a DTX entry is marked as 'CORRUPTED', it means that former DTX resync logic cannot find enough DTX participants to decide the DTX status. Now, in this cycle of DTX resync, we have more participants to claim the DTX status as 'PREPARED', then it is possible to decide its status this time. But if still cannot (processed via dtx_verify_groups()), then it will be handled as 'CORRUPTED' again, harmless.